components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/DASRestClient.java

```diff
@@ -0,0 +1,256 @@
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+import org.apache.http.protocol.HttpContext;
+import org.wso2.carbon.apimgt.usage.client.bean.FirstAccessRequestSearchBean;
+import org.wso2.carbon.apimgt.usage.client.bean.Result;
+import org.wso2.carbon.apimgt.usage.client.bean.SearchRequestBean;
+import org.wso2.carbon.apimgt.usage.client.bean.TableExistResponseBean;
+import org.wso2.carbon.apimgt.usage.client.util.RestClientUtil;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.lang.reflect.Type;
+import java.util.List;
+
+/**
+ * This class contain the actual Client side implementation for the DAS rest client.
+ */
+public class DASRestClient {
+    private CloseableHttpClient httpClient;
+    private String dasUrl;
+    private String user;
+    private String pass;
+    private final Gson gson = new Gson();
+    private static final Log log = LogFactory.getLog(DASRestClient.class);
+
+    /**
+     * get instance providing DAS configuration
+     *
+     * @param url  DAS rest api location
+     * @param user DAS rest api username
+     * @param pass DAs rest api password
+     */
+    public DASRestClient(String url, String user, String pass) {
+        PoolingHttpClientConnectionManager cm = new PoolingHttpClientConnectionManager();
+        httpClient = HttpClients.custom().setConnectionManager(cm).build();
+        this.dasUrl = url;
+        this.user = user;
+        this.pass = pass;
+
+    }
+
+    /**
+     * Do a post request to the DAS REST
+     *
+     * @param json lucene json request
+     * @param url  DAS rest api location
+     * @return return the HttpResponse after the request sent
+     * @throws IOException throw if the connection exception occur
+     */
+    CloseableHttpResponse post(String json, String url) throws IOException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Sending Lucene Query : " + json);
+        }
+
+        HttpPost postRequest = new HttpPost(url);
+        HttpContext context = HttpClientContext.create();
+
+        //get the encoded basic authentication
+        String cred = RestClientUtil.encodeCredentials(this.user, this.pass);
+        postRequest.addHeader(APIUsageStatisticsClientConstants.HTTP_AUTH_HEADER_NAME,
+                APIUsageStatisticsClientConstants.HTTP_AUTH_HEADER_TYPE + ' ' + cred);
+        StringEntity input = new StringEntity(json);
+        input.setContentType(APIUsageStatisticsClientConstants.APPLICATION_JSON);
+        postRequest.setEntity(input);
+
+        CloseableHttpResponse response;
+        try {
+            //send the request
+            response = httpClient.execute(postRequest, context);
+        } finally {
+            //            httpClient.getConnectionManager().shutdown();
```

Comment:
it is close just after the response is read, in next immediate method. and unwanted try block removed in commit : https://github.com/wso2/carbon-apimgt/commit/9619da2271b49f37ef93456b6f8a8ac235f685ea